static/app/views/issueList/customViewsHeader.tsx

```diff
@@ -0,0 +1,436 @@
+          },
+          pathname: `/organizations/${organization.slug}/issues/`,
+        }),
+      };
+    }
+  );
+
+  let initialTabKey = viewsToTabs[0].key;
+  if (defaultView === 'temporary-tab') {
+    initialTabKey = 'temporary-tab';
+  } else if (defaultView) {
+    initialTabKey =
+      viewsToTabs.find(tab => tab.id === defaultView)?.key ?? viewsToTabs[0].key;
+  }
+
+  const [draggableTabs, setDraggableTabs] = useState<Tab[]>(viewsToTabs);
+  const [selectedTabKey, setSelectedTabKey] = useState<string>(initialTabKey);
+  const [tempTab, setTempTab] = useState<Tab | undefined>(
+    initialTabKey === 'temporary-tab'
+      ? {
+          key: 'temporary-tab',
+          label: 'Unsaved',
+          query,
+          querySort: sortParam,
+          to: normalizeUrl({
+            query: {
+              ...queryParams,
+              query,
+              sort: sortParam,
+            },
+            pathname: `/organizations/${organization.slug}/issues/`,
+          }),
+        }
+      : undefined
+  );
+
+  const {mutate: updateViews} = useUpdateGroupSearchViews();
+
+  const debounceUpdateViews = debounce((newTabs: Tab[]) => {
+    if (newTabs) {
+      updateViews({
+        orgSlug: organization.slug,
+        groupSearchViews: newTabs.map(tab => ({
+          id: tab.id,
+          name: tab.label,
+          query: tab.query,
+          querySort: tab.querySort,
+        })),
+      });
+    }
+  }, 1000);
+
+  // Update URL with view's query, sort, and id(?) upon initial render. Ideally
+  // we can remove this after overview.tsx gets overhauled.
+  useEffect(() => {
+    router.replace(
+      normalizeUrl({
+        query: {
+          ...queryParams,
+          query: query,
+          sort: sort,
+          ...(draggableTabs[0]?.id ? {viewId: draggableTabs[0].id} : {}),
+        },
+        pathname: `/organizations/${organization.slug}/issues/`,
+      })
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // Update local tabs when new views are received from mutation request
+  useEffect(() => {
+    setDraggableTabs(
+      draggableTabs.map(tab => {
+        if (!tab.id) {
+          views.forEach(view => {
+            if (
+              tab.query === view.query &&
+              tab.querySort === view.querySort &&
+              tab.label === view.name
+            ) {
+              tab.id = view.id;
+              tab.to = normalizeUrl({
+                query: {
+                  ...queryParams,
+                  query: view.query,
+                  sort: view.querySort,
+                  viewId: view.id,
+                },
+                pathname: `/organizations/${organization.slug}/issues`,
+              });
+            }
+          });
+        }
+        return tab;
+      })
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [views]);
+
+  // Set parent component's bottem border to dashed if temp view is selected
+  useEffect(() => {
```

Comment:
The parent component should be able to look at the URL to see if a temp tab is active or not